core/module/used_refinements_spec.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../../spec_helper'
+
+describe "Module.used_refinements" do
+  ruby_version_is "3.2" do
+    it "returns list of all refined modules used in current scope" do
+      refined_integer = nil
+      refined_string = nil
+      refined_items = []
+
+      r1 =  Module.new do
+        refine Integer do
+          refined_integer = self
+        end
+      end
+
+      r2 = Module.new do
+        refine String do
+          refined_string = self
+        end
+      end
+
+      Module.new do
+        using r1
+        using r2
+
+        refined_items = Module.used_refinements
+      end
+
+      refined_items.should include(refined_integer, refined_string)
+    end
```

Comment:
suggestion: I would also consider `ScratchPad` instead of local variable `refined_items`.

Reply:
**minor:** Also `include` isn't strict check enough. We should check that only these two objects are returned.